core/app/abstract-ui/src/lib/app-abstract-ui.module.ts

```diff
@@ -0,0 +1,25 @@
+import { APP_INITIALIZER, NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { ComponentLoaderDirective } from './directives';
+import { DefaultDialogService, DialogService, ErrorHandlerService } from './services';
+import { ErrorDetailComponent } from './components/error-detail/error-detail.component';
+import { RouterModule } from '@angular/router';
+import { AppComponent } from './components';
+
+const components_directives = [AppComponent, ComponentLoaderDirective];
+
+@NgModule({
+  declarations: [...components_directives, ErrorDetailComponent],
+  imports: [CommonModule, RouterModule],
+  exports: [...components_directives, RouterModule],
+  providers: [
+    { provide: DialogService, useClass: DefaultDialogService },
+    {
+      provide: APP_INITIALIZER,
+      multi: true,
+      useFactory: () => () => {},
```

Comment:
It instantiates an ErrorHandlerService. Otherwise, it might not be used anywhere, so no instance will ever be created (even trea-shaking this from the final app). I agree, it does feel 'hacky'. Would there be a better way?

Reply:
As I see it the `deps` only makes sure the factory gets the `ErrorHandlerService` from the dependency tree and does nothing to instantiate it?

Reply:
It does.